bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormStructureParserImpl.java

```diff
@@ -117,14 +117,16 @@ private String getFormContainerPath(Resource resource) {
 
     public String getFormDefinition() {
         String result = null;
+        if (request != null) {
+            request.setAttribute(FormConstants.X_ADOBE_FORM_DEFINITION, FormConstants.FORM_DEFINITION_SUBMISSION);
```

Comment:
why are we setting this always ? shouldn't it be based on the header ?

Reply:
@vdua , 
the FormStructureParser is only used via FranklinDeliveryServlet, I thought we can keep always assume this call is for publishing form, but i think this api can be used in future for generically getting the form definition so will add the change of adding the attribute only when this header is set.